lib/Doctrine/ORM/Internal/Hydration/SimpleObjectHydrator.php

```diff
@@ -134,6 +136,11 @@ protected function hydrateRowData(array $sqlResult, array &$result)
 
             // Prevent overwrite in case of inherit classes using same property name (See AbstractHydrator)
             if ( ! isset($data[$fieldName]) || ! $valueIsNull) {
+                // If we have inheritance in resultset, make sure the field belongs to the correct class
+                if (isset($cacheKeyInfo['discriminatorValues']) && ! in_array($discrColumnValue, $cacheKeyInfo['discriminatorValues'], true)) {
```

Comment:
Its probably better to test for `$discrColumnValue !== null` here first, because its a more performant check, or?

Reply:
maybe overkill, isset is fast enough. I am a bit worried about this check in the happy hotpath though, but i see no other option.